sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyProperties.cs

```diff
@@ -39,22 +39,22 @@ public KeyProperties(string name)
         /// <summary>
         /// Name of the key.
         /// </summary>
-        public string Name { get; private set; }
+        public string Name { get; internal set; }
```

Comment:
See above comment. I don't think we should open these up. While it's still non-public, we already have a pretty consistent pattern how to deal with this (i.e. `ParseId`).

Reply:
added issue https://github.com/Azure/azure-sdk-for-net/issues/7845 to refactor how Ids are parsed holistically